be/src/exprs/table_function/table_function_factory.cpp

```diff
@@ -92,14 +85,32 @@ const std::unordered_map<std::pair<std::string, bool>, std::function<TableFuncti
                 {{"explode_json_array_string", true}, VExplodeJsonArrayStringCreator},
                 {{"explode_bitmap", true},
                  TableFunctionCreator<vectorized::VExplodeBitmapTableFunction>()},
-                {{"explode", true}, VExplodeCreator},
-                {{"explode_outer", true}, VExplodeOuterCreator}}; // namespace doris
+                {{"explode", true}, TableFunctionCreator<vectorized::VExplodeTableFunction> {}}};
+
+Status TableFunctionFactory::get_fn(std::string fn_name, bool is_vectorized, ObjectPool* pool,
+                                    TableFunction** fn) {
+    bool is_outer = false;
+
+    auto match_suffix = [](const std::string& name, const std::string& suffix) -> bool {
+        return name.substr(name.length() - suffix.length()) == suffix;
```

Comment:
check length at first, maybe name.length() < suffix.length()

Reply:
I think maybe it's  not necessary to do the check here?
http://www.cplusplus.com/reference/string/string/substr/
`Number of characters to include in the substring (if the string is shorter, as many characters as possible are used).`